pkg/services/ngalert/provisioning/templates.go

```diff
@@ -63,11 +63,6 @@ func (t *TemplateService) SetTemplate(ctx context.Context, orgID int64, tmpl def
 		revision.cfg.TemplateFiles = map[string]string{}
 	}
 	revision.cfg.TemplateFiles[tmpl.Name] = tmpl.Template
-	tmpls := make([]string, 0, len(revision.cfg.TemplateFiles))
```

Comment:
AFAIU the config still expects the template file "names" to be listed in the config under `Templates`, even if the templates themselves are passed in memory, or it won't be able to load them. Is something else expanding this back out? (or, did this change in alertmanager?)

Reply:
Ah, I think I misunderstood you entirely. If you mean upstream prometheus alertmanager needing the `Templates` file names then I don't think it does, or at least this PR appears to be working as intended. When upstream is ran through main then it relies on the `configCoordinator` to subscribe to config changes and would likely require the `Templates` field, but we don't use main and instead piece together an alertmanager part-by-part in `grafana/alerting` [GrafanaAlertmanager.ApplyConfig](https://github.com/grafana/alerting/blob/75cd41d40a70a95444b3bef75fe03f810c585cf7/notify/grafana_alertmanager.go#L346) and manually call `Run()` on the `Dispatcher` and `Inhibitor`.